guidelines/monai-application-package.md

```diff
@@ -1,498 +1,588 @@
 
-- Application packages will be deployed in a variety of environments.
+- The scalability of a multi-fragment application based on Holoscan SDK is outside the scope of this document.
+
+- A MONAI Application Package may contain a classical application (non-fragment based), a single-fragment application, or a multi-fragment application.
+
+- Application packages are expected to be deployed in a variety of environments.
 
   > [Note]
   > See [MONAI Operating Environments](monai-operating-environments.md) for additional information about supported environments.
 
+## Requirements
 
-##  Requirements
+The following are requirements that need to be met by the MAP specification to be considered complete and approved.
 
-The following are requirements which need to be met by the MAP specification to be considered complete and approved.
+### Single Artifact
 
+- A MAP SHALL comprise a single container meeting the minimum requirements this document sets forth.
+- A MAP SHALL be a containerized application to maximize the portability of its application.
 
-### Contains an Application
+### Self-Describing
 
-A MAP SHALL contain an executable application which supports the primary function of the MAP, and provide sufficient information to execute the application as intended.
+- A MAP MUST be self-describing and provide a mechanism for extracting its description.
+  - A MAP SHALL provide a method to print the metadata files to the console.
+  - A MAP SHALL provide a method to copy the metadata files to the user-specified directory.
+- The method of description SHALL be in a machine-readable and writable format.
+- The method of description SHALL be in a human-readable format.
+- The method of description SHOULD be a human writable format.
+- The method of description SHALL be declarative and immutable.
+- The method of description SHALL provide the following information about the MAP:
+  - Execution requirements such as dependencies and restrictions.
+  - Resource requirements include CPU cores, system memory, and GPU availability.
 
+### Runtime Characteristics of the MAP
 
-### Single Artifact
+- A MAP SHALL execute the application packaged within when it is started with no specific arguments.
+- A MAP SHALL be a service or an application so an external agent can manage its lifecycle.
 
-A MAP SHALL be comprised of a single container which meets the minimum requirements set forth by this document.
+### IO Specification
 
+- A MAP SHALL provide information about its expected inputs such that an external agent can determine if the MAP can receive a workload.
+- A MAP SHALL provide sufficient information about its outputs so that an external agent knows how to handle the results.
 
-### Self-Describing
+### Local Execution
 
-A MAP MUST be self-describing and provide a mechanism for extracting the its description.
+A MAP MUST be in a format that supports local execution in a development environment.
```

Comment:
Are we saying that the application must be executable in one of the supported dev environments or all of them?

Reply:
I would say a development environment can be anything which means everything listed in the referenced doc.

cc @dbericat

Reply:
May need to qualify it somewhat. MAP's target runtime env is expected to be compatible with dev env for this to happen. E.g. MAP for ARM AArch64 cannot be executed in a dev env on X86.

Reply:
We should make sure a MAP runs on any of the supported operating environments listed here - https://github.com/Project-MONAI/monai-deploy/blob/main/guidelines/MONAI-Operating-Environments.md

We may need to update that guideline next to reflect the latest changes.